code/modules/hydroponics/gene_modder.dm

```diff
@@ -282,9 +295,9 @@
 					repaint_seed()
 				if("extract")
 					if(disk && !disk.read_only)
-						disk.gene = G
-						if(istype(G, /datum/plant_gene/core/potency))
-							var/datum/plant_gene/core/gene = G
+						disk.gene = G.Copy()
+						if(istype(disk.gene, /datum/plant_gene/core/potency))
+							var/datum/plant_gene/core/gene = disk.gene
```

Comment:
This highlights the importance of properly clearing all references and passing things through `qdel` as much as possible, even if it adds a TINY bit of overhead.

It was curious as to why only some of the time genes wouldn't GC...and this is why; basically it would use the extract genes reference and put it on the disk, then immediately delete the seed that held the gene---and that's the problem; since the gene in the seed was still being held as a reference in the disk, the seeds would fail to GC, as would the gene you extracted.

Making the disk hold a new instance (with copied stats) of the seeds gene? Bingo, problem solved.